src/coreclr/vm/gc_unwind_x86.inl

```diff
@@ -0,0 +1,3884 @@
+
+        /* If we're past the "push ebp", adjust ESP to pop EBP off */
+
+        if  (curOffs == (offset + 1))
+            pContext->SP += sizeof(TADDR);
+
+        /* Stack pointer points to return address */
+
+        SetRegdisplayPCTAddr(pContext, (TADDR)pContext->SP);
+
+        /* EBP and callee-saved registers still have the correct value */
+
+        return;
+    }
+
+    // We are atleast after the "push ebp; mov ebp, esp"
+
+    offset = SKIP_MOV_REG_REG(methodStart,
+                SKIP_PUSH_REG(methodStart, offset));
+
+    /* At this point, EBP has been set up. The caller's ESP and the return value
+       can be determined using EBP. Since we are still in the prolog,
+       we need to know our exact location to determine the callee-saved registers */
+
+    const unsigned curEBP = GetRegdisplayFP(pContext);
+
+    if (updateAllRegs)
+    {
+        PTR_DWORD pSavedRegs = PTR_DWORD((TADDR)curEBP);
+
+        /* make sure that we align ESP just like the method's prolog did */
+        if  (info->doubleAlign)
+        {
+            // "and esp,-8"
+            offset = SKIP_ARITH_REG(-8, methodStart, offset);
+            if (curEBP & 0x04)
+            {
+                pSavedRegs--;
+#ifdef _DEBUG
+                if (dspPtr) printf("EnumRef: dblalign ebp: %08X\n", curEBP);
+#endif
+            }
+        }
+
+        /* Increment "offset" in steps to see which callee-saved
+           registers have been pushed already */
+
+        for (unsigned i = 0; i < STRING_LENGTH(CALLEE_SAVED_REGISTERS_MASK); i++)
+        {
+            RegMask regMask = CALLEE_SAVED_REGISTERS_MASK[i];
+            _ASSERTE(regMask != RM_EBP);
+
+            if ((info->savedRegMask & regMask) == 0)
+                continue;
+
+            if (InstructionAlreadyExecuted(offset, curOffs))
+            {
+                SetLocation(pContext, i, PTR_DWORD(--pSavedRegs));
+            }
+
+            // "push reg"
+            offset = SKIP_PUSH_REG(methodStart, offset) ;
+        }
+
+        TRASH_CALLEE_UNSAVED_REGS(pContext);
+    }
+
+    /* The caller's saved EBP is pointed to by our EBP */
+
+    pContext->SetEbpLocation(PTR_DWORD((TADDR)curEBP));
+    pContext->SP = DWORD((TADDR)(curEBP + sizeof(void *)));
+
+    /* Stack pointer points to return address */
+
+    SetRegdisplayPCTAddr(pContext, (TADDR)pContext->SP);
+}
+
+/*****************************************************************************/
+
+bool UnwindEbpDoubleAlignFrame(
+        PREGDISPLAY     pContext,
+        hdrInfo        *info,
+        PTR_CBYTE       table,
+        PTR_CBYTE       methodStart,
+        DWORD           curOffs,
+        IN_EH_FUNCLETS_COMMA(PTR_CBYTE       funcletStart)
+        IN_EH_FUNCLETS_COMMA(bool            isFunclet)
+        bool            updateAllRegs
+        NOT_IN_NATIVEAOT_PRE_COMMA(StackwalkCacheUnwindInfo  *pUnwindInfo)) // out-only, perf improvement
```

Comment:
Original prototype of the method was
```
bool UnwindEbpDoubleAlignFrame(
        PREGDISPLAY     pContext,
        EECodeInfo     *pCodeInfo,
        hdrInfo        *info,
        PTR_CBYTE       table,
        PTR_CBYTE       methodStart,
        DWORD           curOffs,
        unsigned        flags,
        StackwalkCacheUnwindInfo  *pUnwindInfo) // out-only, perf improvement
```

The `pCodeInfo` parameter was only relevant for funclets which is now expanded into two parameters.